error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/ImplicitBlockingFluxOperation.java

```diff
@@ -0,0 +1,130 @@
+package tech.picnic.errorprone.bugpatterns;
+
+import static com.google.errorprone.BugPattern.LinkType.CUSTOM;
+import static com.google.errorprone.BugPattern.SeverityLevel.ERROR;
+import static com.google.errorprone.BugPattern.StandardTags.LIKELY_ERROR;
+import static com.google.errorprone.matchers.Matchers.anyOf;
+import static com.google.errorprone.matchers.method.MethodMatchers.instanceMethod;
+import static tech.picnic.errorprone.bugpatterns.util.Documentation.BUG_PATTERNS_BASE_URL;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.MethodInvocationTreeMatcher;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.fixes.SuggestedFixes;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.suppliers.Suppliers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import tech.picnic.errorprone.bugpatterns.util.ThirdPartyLibrary;
+
+/**
+ * A {@link BugChecker} that flags usages of Flux methods that are documented to block, but that
+ * behaviour is not apparent from their signature.
+ *
+ * <p>The alternatives suggested are explicitly blocking operators, highlighting actual behavior,
+ * not adequate for automatic replacements, they need manual review.
+ */
+@AutoService(BugChecker.class)
+@BugPattern(
+    explanation =
+        "`Flux#toStream` and `Flux#toIterable` are documented to block, "
+            + "but this is not apparent from the method signature; "
+            + "please make sure that they are used with this in mind.",
+    summary = "Accidental blocking of `Flux` with convenience method.",
+    link = BUG_PATTERNS_BASE_URL + "ImplicitBlockingFluxOperation",
+    linkType = CUSTOM,
+    severity = ERROR,
+    tags = LIKELY_ERROR)
+public class ImplicitBlockingFluxOperation extends BugChecker
+    implements MethodInvocationTreeMatcher {
+  private static final long serialVersionUID = 1L;
+  private static final Matcher<ExpressionTree> FLUX_TO_ITERABLE =
```

Comment:
Will push a change for this one. Now the check I have at the bottom of the class is not so nice, will come back to that. Maybe indeed we need this setup but utilizing the `#namedAnyOf` makes a bit more sense to me. Maybe there is a easier way of checking whether we are making a `SuggestedFix` for the `toIterable` variant or not.